app/src/events.c

```diff
@@ -0,0 +1,83 @@
+
+#include <zephyr.h>
+#include <drivers/behavior.h>
+#include <zmk/behavior.h>
+#include <zmk/events.h>
+#include <sys/util.h>
+
+#define DT_DRV_COMPAT zmk_behavior_global
+#define GLOBAL_BEHAVIOR_LEN DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)
+
+#define LABEL_ENTRY(i) DT_INST_LABEL(i),
+static const char *global_behaviors[] = {
+    DT_INST_FOREACH_STATUS_OKAY(LABEL_ENTRY)
```

Comment:
This is how we pick up all behaviors w/ the extra "zmk,behavior-global" compatible value, and pick those to notify of global events.